ev3/legoev3_ports_out.c

```
@@ -677,16 +677,41 @@ static int ev3_output_port_set_device(void *context, const char *device_name)
 {
 	struct ev3_output_port_data *data = context;
 	struct lego_device *new_motor;
+	struct ev3_motor_platform_data pdata;
 
-	if (data->motor_type != MOTOR_TACHO)
+	switch (data->out_port.mode) {
+	case EV3_OUTPUT_PORT_MODE_AUTO:
 		return -EOPNOTSUPP;
+		break;
+	case EV3_OUTPUT_PORT_MODE_TACHO_MOTOR:
+		if (data->motor_type != MOTOR_TACHO)
+			return -EOPNOTSUPP;
+
+		pdata.tacho_int_gpio = data->gpio[GPIO_PIN5_INT].gpio;
+		pdata.tacho_dir_gpio = data->gpio[GPIO_PIN6_DIR].gpio;
+
+		break;
+	case EV3_OUTPUT_PORT_MODE_DC_MOTOR:
+		return -EOPNOTSUPP;
+		break;
+	case EV3_OUTPUT_PORT_MODE_LED:
+		return -EOPNOTSUPP;
+		break;
+	case EV3_OUTPUT_PORT_MODE_RAW:
+		return -EOPNOTSUPP;
+		break;
+	default:
+		WARN_ON("Unknown mode.");
+		break;
```

Shouldn't we return an error in this case?